src/pcrlock/pcrlock.c

```diff
@@ -2769,7 +2769,7 @@ static int verb_lock_raw(int argc, char *argv[], void *userdata) {
                         return log_error_errno(errno, "Failed to open '%s': %m", argv[1]);
         }
 
-        r = read_full_stream(f ?: stdin, &data, &size);
+        r = read_full_stream_full(f ?: stdin, NULL, UINT64_MAX, SIZE_MAX, READ_FULL_FILE_LARGE_FILE, &data, &size);
```

Comment:
I very much don't like this solution. Reading a 64MB file into memory is a bit sloppy but OK on most systems. But reading 1GB files into memory is not right. And the 1GB limit is arbitrary too, I'm sure that there'll be people who'll want it raised because they embed nvidia firmware and whatnot. The internals should be reworked to use a streaming API to hash the file.

Reply:
> The internals should be reworked to use a streaming API to hash the file.

I was thinking on creating a `make_pcrlock_record` variant that will accept a fd and use it only in those two verbs. Would that be accepted?

Reply:
that sounds fine